components/nexusoperations/workflow/commands.go

```diff
@@ -201,17 +197,17 @@ func (ch *commandHandler) HandleCancelCommand(
 		}
 		he.UserMetadata = command.UserMetadata
 	})
-	return coll.Transition(nodeID, func(o nexusoperations.Operation) (hsm.TransitionOutput, error) {
-		output, err := o.Cancel(node, event.EventTime.AsTime())
-		// Cancel spawns a child Cancelation machine, if that machine already exists we got a duplicate cancelation request.
-		if errors.Is(err, hsm.ErrStateMachineAlreadyExists) {
-			return output, workflow.FailWorkflowTaskError{
-				Cause:   enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_REQUEST_CANCEL_NEXUS_OPERATION_ATTRIBUTES,
-				Message: fmt.Sprintf("cancelation was already requested for an operation with scheduled event ID of %d", attrs.ScheduledEventId),
-			}
+
```

Comment:
not related to this change: I think it's possible to encounter ErrStateMachineNotFound on L168 if OperationCompletedEvent is buffered? Could be confusing to user I guess.

Reply:
I'll see how we deal with this for activities.

Reply:
I saw what we do for activities, we look up any close events in the buffer.
This isn't an issue yet because we don't delete the state machine but it may become an issue later.

Reply:
I'll add a comment.